### PR TITLE
Update comment about minimum Numpy version

### PR DESCRIPTION
**Pull request by larrybradley**
### Description

This pull request addresses the comment in https://github.com/astropy/astropy/pull/12146#issuecomment-1503986279.

The conditional block mentioned is still required for Numpy 1.21.  The issue of `np.median` dropping units for NaN output was fixed in Numpy 1.22.
 
https://github.com/astropy/astropy/issues/12165 was essentially fixed upstream by Numpy 1.22, so I think we can close that issue.

### CODE REVIEW

File: astropy/stats/funcs.py [hunk 1]
```diff
@@ -815,8 +815,7 @@ def median_absolute_deviation(data, axis=None, func=None, ignore_nan=False):
     # np.nanmedian has `keepdims`, which is a good option if we're not allowing
     # user-passed functions here
     data_median = func(data, axis=axis)
-    # this conditional can be removed after this PR is merged:
-    # https://github.com/astropy/astropy/issues/12165
+    # this conditional can be removed for minimum supported Numpy >= 1.22
```
Comment (pllim) [inline astropy/stats/funcs.py:818]:
```suggestion
    # this conditional can be removed for minimum supported Numpy >= 1.22 (NUMPY_LT_1_22)
```

**Review by pllim — COMMENTED** [review]
Thanks! Can you please accept my suggestions so we can find it again when we bump numpy minversion in the future?

**Review by mhvk — CHANGES_REQUESTED** [review]
Was reviewing this and then realized this really should be fixed in `units`, not worked around in `stats` - see #14635.

Assuming that works, this can be closed; for now marking as "request changes" so we don't merge prematurely.

**Comment by pllim** [comment]
Superseded by #14635

**Pull request closed without merging**